tests/data/test_packing.py

```diff
@@ -194,28 +193,57 @@ def test_auto_packing_with_streaming_dataloader(tmp_path: Path):
         'prefetch_factor': None,
         'persistent_workers': False,
         'timeout': 0,
-    })
-
-    loader = build_finetuning_dataloader(
-        **cfg,
-        tokenizer=tokenizer,
-        device_batch_size=6,
-    ).dataloader
+    }
 
-    batch_ix = 0
-    for _ in loader:
-        batch_ix += 1
-        if batch_ix >= 3:
-            break
+    remote_cfg = DictConfig({
+        **base_cfg,
+        'dataset': {
+            **base_cfg['dataset'],
+            'remote': remote_dir,
+            'local': local_dir,
+        },
+    })
 
-    assert isinstance(loader, DataLoader)
-    assert isinstance(loader.dataset, StreamingFinetuningDataset)
-    assert loader.dataset.packing_ratio is not None
-    assert isinstance(loader.batch_size, int)
-    assert loader.dataset.packing_ratio == int(loader.batch_size / 6)
+    streams_cfg = DictConfig({
+        **base_cfg,
+        'dataset': {
+            **base_cfg['dataset'],
+            'streams': {
+                'stream_with_remote': {
+                    'remote': remote_dir,
+                    'local': local_dir,
+                },
+                'stream_without_remote': {
+                    'local': remote_dir,
+                },
+            },
+        },
+    })
 
-    state_dict = loader.dataset.state_dict(num_samples=2, from_beginning=False)
-    assert state_dict['sample_in_epoch'] == 2 * loader.dataset.packing_ratio
+    for cfg in [remote_cfg, streams_cfg]:
```

Comment:
rather than adding a for loop to the test, please use `@pytest.mark.parametrize`. That way each test can fail individually instead of only all together